modules/unavailability-of-the-kernel-driver.adoc

```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * troubleshooting/commonly-occurring-error-conditions.adoc
+:_module-type: CONCEPT
+[id="unavailability-of-the-kernel-driver_{context}"]
+= Unavailability of the kernel driver
+
+The Collector will determine if it has a kernel driver for the node's particular kernel version. The Collector makes an attempt to download a driver from the Sensor after first searching the local storage for one with the appropriate version and type. The following logs indicate that neither a local kernel driver nor a driver from the Sensor are present:
+
+[source,terminal]
+----
+Collector Version: 3.12.0
+OS: Alpine Linux v3.14
+Kernel Version: 5.10.109-0-virt
+Starting StackRox Collector...
+[INFO    2022/10/13 13:32:57] Hostname: 'alpine'
+[...]
+[INFO    2022/10/13 13:32:57] Sensor configured at address: sensor.stackrox.svc:9999
+[INFO    2022/10/13 13:32:57] Attempting to connect to Sensor
+[INFO    2022/10/13 13:32:57] Successfully connected to Sensor.
+[INFO    2022/10/13 13:32:57] Module version: 2.2.0
+[INFO    2022/10/13 13:32:57] Attempting to find kernel module - Candidate kernel versions:
+[INFO    2022/10/13 13:32:57] 5.10.109-0-virt
+[INFO    2022/10/13 13:32:57] Local storage does not contain collector-5.10.109-0-virt.ko
+[...]
+[INFO    2022/10/13 13:32:57] Attempting to download kernel object from https://sensor.stackrox.svc/kernel-objects/2.2.0/collector-5.10.109-0-virt.ko.gz
+[WARNING 2022/10/13 13:32:58] [Throttled] Unexpected HTTP request failure (HTTP 404)
+[WARNING 2022/10/13 13:33:08] [Throttled] Unexpected HTTP request failure (HTTP 404)
+[WARNING 2022/10/13 13:33:18] [Throttled] Unexpected HTTP request failure (HTTP 404)
+[WARNING 2022/10/13 13:33:29] [Throttled] Unexpected HTTP request failure (HTTP 404)
+[WARNING 2022/10/13 13:33:35] Attempted to download collector-5.10.109-0-virt.ko.gz 30 time(s)
+[WARNING 2022/10/13 13:33:35] Failed to download from collector-5.10.109-0-virt.ko.gz
+[WARNING 2022/10/13 13:33:35] Unable to download kernel object collector-5.10.109-0-virt.ko to /module/collector.ko.gz
+[ERROR   2022/10/13 13:33:35] Error getting kernel object: collector-5.10.109-0-virt.ko
+[INFO    2022/10/13 13:33:35]
+[INFO    2022/10/13 13:33:35] == Collector Startup Diagnostics: ==
+[INFO    2022/10/13 13:33:35]  Connected to Sensor?       true
+[INFO    2022/10/13 13:33:35]  Kernel driver available?   false
+[INFO    2022/10/13 13:33:35]  Driver loaded into kernel? false
+[INFO    2022/10/13 13:33:35] ====================================
+[INFO    2022/10/13 13:33:35]
+[FATAL   2022/10/13 13:33:35]  No suitable kernel object downloaded for kernel 5.10.109-0-virt
+----
+
+
+As a result of missing a  driver, the Collector will enter `CrashLoopBackOff`. The attempts to locate the module locally will be displayed in the logs first, followed by any efforts to download the driver from the sensor. The 404 errors indicate that the node's kernel does not have a kernel driver.
```

Comment:
```suggestion
As a result of missing a driver, the Collector enters the `CrashLoopBackOff` state. The attempts to locate the module locally are displayed in the logs first, followed by any efforts to download the driver from the sensor. The 404 errors indicate that the node's kernel does not have a kernel driver.
```

Reply:
We should use code comments to highlight this info.

Reply:
Also avoid using passive voice:
- the logs display attempts to ...

Reply:
ack